components/developer/rust/Makefile

```diff
@@ -27,43 +27,49 @@
 # Copyright 2023, Friedrich Kink
 #
 
+# This build is a mixture of configure and python style
+# it starts with configure for the initial setup 
+# the compiler build itself is intensively using python tools
+# therefor this patially python settings
+
 BUILD_BITS=		64
-USE_OPENSSL11=		yes
+OPENSSL_DEFAULT=	3.1
+TEST_STYLE=		none
 
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		rustc
-COMPONENT_VERSION=	1.70.0
+COMPONENT_VERSION=	1.71.1
 COMPONENT_SUMMARY=	Rust - Safe, concurrent, practical language
 COMPONENT_PROJECT_URL=	https://www.rust-lang.org
 COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)-src
 COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
-COMPONENT_ARCHIVE_HASH=	sha256:b2bfae000b7a5040e4ec4bbc50a09f21548190cb7570b0ed77358368413bd27c
+COMPONENT_ARCHIVE_HASH=	sha256:6fa90d50d1d529a75f6cc349784de57d7ec0ba2419b09bde7d335c25bd4e472e
 COMPONENT_ARCHIVE_URL=	https://static.rust-lang.org/dist/$(COMPONENT_ARCHIVE)
 COMPONENT_FMRI=		developer/lang/rustc
 COMPONENT_CLASSIFICATION=	Development/Other Languages
 COMPONENT_LICENSE=	MIT or Apache-2.0
 COMPONENT_LICENSE_FILE=	LICENSE-APACHE
 
-RUST_PRE_VERSION=	$(COMPONENT_VERSION)
 RUST_BOOTSTRAP_PATH=	$(BUILD_DIR)/$(MACH64)
 RUST_ARCH=		x86_64-unknown-illumos
 
 TEST_TARGET=		$(NO_TESTS)
 
 include $(WS_MAKE_RULES)/common.mk
+include $(WS_MAKE_RULES)/setup.py.mk
+
+CLANG_VERSION=		17
 
 # Need some help to pick the correct ar binary
 GNUAR=$(GNUBIN)/ar
 
 # Put the bits cargo downloads in a private directory.  This could be cached
 # somewhere more permanent, but it's important to make sure that a person's
 # $HOME/.cargo isn't used.
-RUST_VERSION=           $(COMPONENT_VERSION)
-RUSTUP_HOME=            $(@D)
-CARGO_HOME=             $(RUSTUP_HOME)
+CARGO_HOME=             $(@D)/.cargo
 
-PATH=                   $(CARGO_HOME)/bin:$(CLANG_PREFIX)/bin:$(GCC_BINDIR):$(PATH.gnu)
+PATH=                   $(PATH.gnu):$(CLANG_PREFIX)/bin:$(GCC_BINDIR)
```

Comment:
`$(CLANG_PREFIX)/bin` and `$(GCC_BINDIR)` is not needed here.  It is already a part of the `$(PATH.gnu)` macro.